### PR TITLE
slogdet of ivy functional linear algebra reformatted with examples in docstring

### PR DESCRIPTION
**Pull request by ash-ishh**

### CODE REVIEW

**Comment by ash-ishh** [comment]
Close #4450

**Comment by ash-ishh** [comment]
TODO:

1 - Container representation issue
In ivy/container/base.py for named tuple follwing condition is executed

```
                elif (
                    isinstance(v, (list, tuple))
                    and v
                    and (self._ivy.is_native_array(v[0]) or isinstance(v[0], ivy.Array))
                ):
                    rep = (
                        "list[{}]".format(len(v)),
                        type(v[0]),
                        "shape=",
                        list(v[0].shape),
                    )
```

Which results in output with just len, type and shape
```
    {
        a: (list[2], <class ivy.array.array.Array> shape=[3])
    }
```
Should we print elements as well? will it affect existing desing decision?

In case of slogdet it will look something like this
```
    {
        a: ('slogdet(sign=ivy.array(-1.), logabsdet=ivy.array(0.69314718))', list[2], <class ivy.array.array.Array> shape=[3])
    }
```
2 - Type hint issue

In the backend implementations of slogdet `collections namedtuple` is used for return but in other parts of code `typing NamedTuple` is used
Should be fix one class for return? if yes which one
Difference between them: https://stackoverflow.com/questions/50766461/whats-the-difference-between-namedtuple-and-namedtuple

**Comment by ash-ishh** [comment]
add_reformatting_checklist_linalg

**Comment by ash-ishh** [comment]
add_reformatting_checklist_linear_algebra

**Comment by ash-ishh** [comment]
add_reformatting_checklist_linear_algebra

**Comment by ash-ishh** [comment]
Hi @abdrahmandiab @djl11 :wave: need your views on how should I proceed with the above 3 constraints

1 - Should I update container repr for a more descriptive namedtuple as part of this pr? or should I create the separate issue and let the example be in the current format only
2 - Should I go ahead with `collections namedtuple` for the return type hint?
3 - Should I remove optional `out` from ivy slogdet since backends are returning namedtuples?

Thanks.

**Comment by ash-ishh** [comment]
add_reformatting_checklist_linear_algebra

**Comment by abdrahmandiab** [comment]
Hi @ash-ishh, I am going to start the review process for your PR now, but for future reference, please do request a review when your work is ready for evaluation. 😄 

Thanks!

**Comment by abdrahmandiab** [comment]
Hi @ash-ishh, 
I've asked Daniel about your queries regarding the `namedtuple` stuff and either me or him will be returning to you with a response on that shortly!

In the meantime, I would recommend checking out the Python Array API page on [`slogdet`](https://data-apis.org/array-api/2021.12/extensions/generated/signatures.linalg.slogdet.html?highlight=slogdet). Feel free to follow the type hints here as this is the standard that Ivy adheres to. Additionally, you can checkout this page for [`torch.slogdet`](https://pytorch.org/docs/stable/generated/torch.linalg.slogdet.html#torch.linalg.slogdet) for an example that implements the `out` argument.
* Note: it appears torch has a specialized dtype `torch.return_types.linalg_slogdet` which is returned when you make a call to slogdet.

Also, while playing around, I've found that the out argument doesn't actually work even if you pass in a correctly named `namedtuple`, but that's something we'll be looking at internally so no worries there! 😅

Examples:
```
>>> from collections import namedtuple
>>> x = ivy.array([[1,2],[3,4]])
>>> n_t = namedtuple('slogdet',['sign','logabsdet'])
>>> z = n_t(ivy.array(0.),ivy.array(0.))
>>> z
slogdet(sign=ivy.array(0.), logabsdet=ivy.array(0.))
>>> ivy.slogdet(x,out=z)
slogdet(sign=ivy.array(-1.), logabsdet=ivy.array(0.69314718))
>>> z
slogdet(sign=ivy.array(0.), logabsdet=ivy.array(0.))
```
*and*
```
>>> x = ivy.array([[1,2],[3,4]])
>>> y = ivy.slogdet([[0.,0.],[0.,0.]])
>>> y
slogdet(sign=ivy.array(0.), logabsdet=ivy.array(-inf))
>>> ivy.slogdet(x,out=y)
slogdet(sign=ivy.array(-1.), logabsdet=ivy.array(0.69314718))
>>> y
slogdet(sign=ivy.array(0.), logabsdet=ivy.array(-inf))
```

**Comment by djl11** [comment]
> Hi @abdrahmandiab @djl11 wave need your views on how should I proceed with the above 3 constraints
> 
> 1 - Should I update container repr for a more descriptive namedtuple as part of this pr? or should I create the separate issue and let the example be in the current format only 2 - Should I go ahead with `collections namedtuple` for the return type hint? 3 - Should I remove optional `out` from ivy slogdet since backends are returning namedtuples?
> 
> Thanks.

1. Don't worry about changing `ivy.Container.__repr__` as part of this PR, but thank you so much for pointing this out, we will incorporate your suggestions (or a similar variant) soon!
2. actually, changing all of these to typing.NamedTuple would be great, the Array API Standard [does not specify](https://github.com/data-apis/array-api/blob/eba54b35b040bb08359fdca7aaeb0340ecb5c87c/spec/API_specification/array_api/linalg.py#L347) one vs the other.
3. Yes please, in Ivy we only intend to support `out` for array returns, and so this does not apply for this function. Again, thanks for pointing this out!

**Comment by djl11** [comment]
Once these changes are made, feel free to review + merge @abdrahmandiab, thanks both!

**Comment by ash-ishh** [comment]
Thank you @djl11 @abdrahmandiab 

PR is ready for review.

**Comment by ash-ishh** [comment]
Hi @abdrahmandiab just a nudge for review and merge.

**Comment by abdrahmandiab** [comment]
Hi @ash-ishh, sorry it took me a bit to review! 😅

There are actually a few failing docstring examples, could you please attempt to fix them, and of course if you need any help I'll be here! 

To checkout these failures, you can run the test_docstrings locally and then search for `slogdet` in the output logs and failures arising from it. Alternatively, at the bottom of this "Conversation" page on the PR, you will find our github check actions. You can check for docstrings failures by clicking on "Details" next to the "test-docstrings / run-docstring-tests (pull_request)" check. 
image

After this, please navigate to the "Run Docstring Tests" job and search the logs for `slogdet`, below is an example! 😃
image


Thanks! 😊

**Comment by ash-ishh** [comment]
@abdrahmandiab All the slogdet docstring tests are passing now.

**Review by abdrahmandiab — CHANGES_REQUESTED** [review]
Hi @ash-ishh, sorry for the late reply, I had wrote this review a few days ago and forgot to submit it haha 😅 

Thanks for fixing the tests, and after these small changes hopefully we can merge right away!
Great job!

File: ivy/container/linear_algebra.py
Comment (abdrahmandiab) [inline ivy/container/linear_algebra.py:1352]: could you please add a `b` element here as well? 😃

File: ivy/functional/ivy/linear_algebra.py
Comment (ash-ishh) [reply ivy/functional/ivy/linear_algebra.py:1409]: Instance methods of slogdet seems valid for both array and container, any specific reason why they are not necessary?

Comment (abdrahmandiab) [reply ivy/functional/ivy/linear_algebra.py:1406]: they are correct and all, but they are already included in the relevant function be it instance or static, and therefore adding them here is just redundant and leads to unnecessary cluttering 😅

File: ivy/container/linear_algebra.py
Comment (abdrahmandiab) [reply ivy/container/linear_algebra.py:1349]: @ash-ishh this comment was marked as resolved even though it was not addressed 😅

**Review by abdrahmandiab — CHANGES_REQUESTED** [review]

File: ivy/functional/ivy/linear_algebra.py
Comment (ash-ishh) [reply ivy/functional/ivy/linear_algebra.py]: Makes sense, I have removed them.

**Comment by abdrahmandiab** [comment]
Hi @ash-ishh, everything looks good to me! :)

Will be merging soon! Thanks for your work! 🚀

**Pull request merged**
